controllers/ArticleController.js

```diff
@@ -441,4 +441,72 @@ export default class ArticleController {
       return next(err);
     }
   }
+
+  /**
+   *
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @method bookmarkArticle
+   * @returns {object} returns the details of the bookmarkedmitem
+   * @static
+   */
+  static async bookmarkArticle(req, res, next) {
+    try {
+      const { id: articleId } = req.params;
+      const { id } = req.user;
+      const bookMarked = await Bookmark.findOne({
+        where: {
+          userId: id,
+          articleId
+        }
+      });
+      if (bookMarked) {
+        await Bookmark.destroy({ where: { userId: id } });
+        return successResponse(res, 200, 'bookmark', { message: 'Article has been removed from bookmarked successfully' });
+      }
+      try {
+        await Bookmark.create({
+          userId: id,
+          articleId,
+        });
+      } catch (error) {
+        return errorResponse(res, 400, {
+          bookmark: 'Something went wrong, unable to bookmark article'
+        });
+      }
+      return successResponse(res, 200, 'bookmark', { message: 'Article has been bookmarked successfully' });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   *
+   * @param {object} req express request object
+   * @param {Object} res express respond object
+   * @param {function} next
+   * @returns {object} All the articles bookmarked by the user
+   */
+  static async getAllBookmark(req, res, next) {
+    try {
+      const { id } = req.user;
+      const { id: userId } = req.params;
+      if (id !== userId) {
+        return errorResponse(res, 401, 'You are not allowed to view this user\'s bookmarks');
```

Comment:
Is `401` the right status code for this?

Reply:
Yes, that means the user is inputing a valid `id`, but the `id` belongs to someone else. So he/she is not authorised.

Reply:
In that case, it is not an authentication error. The user does not have the permission to view another user's bookmarks. Is that still a `401`?

Reply:
Oh yeah, that's true. I would change it now

Reply:
Fixed @mazma1